### DISCUSSION TITLE
Can't get this menu to open

### DISCUSSION THREAD

**Discussion by Rohyn** [evennia/evennia#3679] [Evennia Questions] [answered]
Hey all, I _cannot_ figure out how to get this EvMenu to open for NPC shops. I have it set up exactly like other menus that work perfectly fine, but the evmenu can't seem to find the script it's pointing to, I think, and I have no idea why. I've been chewing on this for almost a week now and could really use some fresh eyes please. Here is the error:

`wares
Traceback (most recent call last):
  File "C:\Users\Owner\Documents\Projects\MUDdev\evennia\evennia\commands\cmdhandler.py", line 629,
in _run_command
    ret = cmd.func()
          ^^^^^^^^^^
  File "C:\Users\Owner\Documents\Projects\MUDdev\ROC\commands\inventory.py", line 1185, in func
    evmenu.EvMenu(caller,
  File "C:\Users\Owner\Documents\Projects\MUDdev\ROC\world\evmenu.py", line 596, in __init__
    self._menutree = self._parse_menudata(menudata)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Owner\Documents\Projects\MUDdev\ROC\world\evmenu.py", line 740, in _parse_menudata
    for key, func in module.__dict__.items()
                     ^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__dict__'. Did you mean: '__dir__'?

An untrapped error occurred.
(Traceback was logged 24-11-17 06:53:09-07).`

Here's the section of the 'wares' command in question:
`		else: # we're at a peddler
			if args:
				caller.message("Syntax: BUY", msg_type="error")
				return
			caller.db.conversing_with = peddler
			evmenu.EvMenu(caller,
						  "world.shops",
						  startnode="menunode_shopfront",
						  allow_quit=True,
						  cmd_on_exit="ql")`

This is the first node in world/shops:
`def menunode_shopfront(caller, raw_input, **kwargs): 
	"This is the top-menu screen."

	if caller.location.db.stockroom: # we're at a shop
		shopname = caller.location.key
		if isinstance(caller.location.db.stockroom, int):
			room = ObjectDB.objects.filter(id=caller.location.db.stockroom)
			caller.location.db.stockroom = room[0]
		caller.ndb._evmenu.stockroom = caller.location.db.stockroom
	else: # we're at a peddler or something
		if caller.db.conversing_with.name[-1] == "s":
			pos = "'"
		else:
			pos = "'s"
		shopname = "%s%s wares" % (caller.db.conversing_with.name, pos)
		caller.ndb._evmenu.stockroom = caller.db.conversing_with.db.stockroom
	#wares = stockroom.contents
	shelves = []
	count = 0
	for item in caller.ndb._evmenu.stockroom.contents:
		if item.db.prof == "shop-shelf":
			shelves.append(item)
		else:
			count += 1

	text = "*** Welcome to %s! ***\n" % shopname    
	text += "   Available shelves (choose a number to inspect);" \
			" LEAVE to exit:"     

	options = []
	options.append({"desc" : "General Wares (%s items)" % str(count),
					"goto" : ("menunode_shopshelf", {'shelf' : 'general'})})
	for shelf in shelves:
		options.append({"desc" : "%s%s (%s items)" % (shelf.db.paint, shelf.name, str(len(shelf.contents))),
						"goto" : ("menunode_shopshelf", {'shelf' : shelf})})
	options.append({"desc" : "Nevermind.",
					"goto" : "menunode_shopend"}) 
	return text, options`

world/evmenu is literally an exact copy of the evennia/utils/evmenu, I plan to make changes but haven't made any of them yet. The lines in question in the error are:
Line 596 in 'class EvMenu'
`        self._menutree = self._parse_menudata(menudata)`
And line 740 is part of this:
`    def _parse_menudata(self, menudata):
        if isinstance(menudata, dict):
            # This is assumed to be a pre-loaded menu tree.
            return menudata
        else:
            # a python path of a module
            module = mod_import(menudata)
            return dict(
                (key, func)
                for key, func in module.__dict__.items()
                if isfunction(func) and not key.startswith("_")
            )`

I'm so frustrated by this because I have multiple other menus that are set up exactly the same way and work perfectly fine! I've checked the spelling over and over and can't figure out why the command doesn't seem to be able to find the world.shops. It's definitely in the right directory, and the shops is listed in the pycache so the game has seen it at some point. I just don't know what else to check or do. If anyone has a suggestion please, please let me know. Thank you!

**Comment by InspectorCaracal** [comment, marked as answer]
What this error means:
```
File "C:\Users\Owner\Documents\Projects\MUDdev\ROC\world\evmenu.py", line 740, in _parse_menudata
for key, func in module.__dict__.items()
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__dict__'. Did you mean: '__dir__'?
```
is that when the EvMenu tried to load your module "world.shops", it failed. The fastest way to identify the issue is to run `evennia shell` in your command line, then type `import world.shops` to see what error you get.

**Reply by Rohyn** [reply]
Thank you SO MUCH! It was a simple issue, a misspelling in one of the imports at the top of the module. Can't believe I spent so long on this, ugh.

**Reply by InspectorCaracal** [reply]
No problem, glad you got it worked out!
